enhancements/monitoring/integration-with-topology-modes.md

```diff
@@ -0,0 +1,90 @@
+---
+title: monitoring-topology-mode
+authors:
+  - "@lilic"
+reviewers:
+  - "monitoring-team"
+  - "???"
+approvers:
+  - "monitoring-team"
+creation-date: 2021-03-01
+last-updated: 2021-03-01
+status: implementable
+see-also:
+  - "/enhancements/cluster-high-availability-mode-api.md"
+---
+
+# Monitoring Stack on Single Replica Topology Mode
+
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The monitoring stack needs to react based on the `infrastructureTopology` field newly added to `infrastructures.config.openshift.io` CRD. Plan is to detect whenever
+cluster is in single replica topology mode and reduce the monitoring stack footprint by adjusting platform monitoring
+instances to be non Highly Available.
+
+## Motivation
+
+In the 4.8 time frame, OpenShift will introduce a new capabilities field in the Infrastructure API that will inform
+operators about the deployment topology. The immediate goal is for operators to adjust the deployment of their operands
+depending on whether the infrastructure is HA (at least 3 nodes) or not (single node).
+
+### Goals
+
+- Reduce monitoring footprint by lowering number of replicas
```

Comment:
This feels a bit like it's mixing 2 goals together. We need to be able to run in a non-HA mode because we may not have a place to put a 2nd or 3rd replica. We may also need to reduce the overall footprint, but that may come on top of the reduction to 1 replica.

Since the second part is called out as a non-goal, perhaps this would be better phrased as "Support monitoring stack with a single replica for non-HA configurations" or something like that?

Reply:
lowering number of replicas may be sufficient, so it seems fine as an initial goal.  If we need more squeezing afterwards (e.g. by lowering scrape periods?  Reducing stored history?  I don't know what the relevant Prom knobs are), we can do that in follow-up work.  If we know right now that we will need those too, seems like it's worth putting them in as goals.  Or we could make the goal something like "monitoring does whatever they need to do to get under $CPU_QUOTA for $FRACTION of a single-node CI run", and leave the details unspecified in the enhancement.

Reply:
Agreed with @dhellmann. To be clear the scope of the enhancement from the monitoring point of view is strictly to support the new SingleReplica mode.